pkg/staticnetworkconfig/generator.go

```diff
@@ -61,6 +61,42 @@ func New(log logrus.FieldLogger) StaticNetworkConfig {
 	}
 }
 
+// TODO: Temporary workaround until the nmstate team resolves and deploys the following issue - https://issues.redhat.com/browse/RHEL-59935
+func (s *StaticNetworkConfigGenerator) injectIPV4V6FieldsIfNeeded(networkYaml string) (string, error) {
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networkYaml), &config)
+	if err != nil {
+		s.log.WithError(err).Errorf("Error unmarshalling yaml")
+		return "", err
+	}
+	Interfaces := config["interfaces"].([]interface{})
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		if val, exists := nic["ipv4"].(map[interface{}]interface{}); exists {
+			if _, exists := val["dhcp"]; !exists {
+				val["dhcp"] = false
+			}
+		}
+		if val, exists := nic["ipv6"].(map[interface{}]interface{}); exists {
+			if _, exists := val["dhcp"]; !exists {
+				val["dhcp"] = false
+			}
+			if _, exists := val["autoconf"]; !exists {
+				val["autoconf"] = false
```

Comment:
I think we should  also check if  dhcp is false before setting auto conf + let add this case to tests so we wont have dhcp = true and autoconf = false

Reply:
`dhcp = true` and `autoconf = false`  is translated to `method=dhcp` in the network manager key file. It is a valid config (stateful IPv6).